pkg/cache/tas_flavor_snapshot.go

```diff
@@ -344,11 +344,16 @@ func (s *TASFlavorSnapshot) buildTopologyAssignmentForLevels(domains []*domain,
 			Values: domain.levelValues[levelIdx:],
 			Count:  domain.state,
 		}
+		usage := make(resources.Requests)
+		for k, v := range singlePodRequest {
+			usage[k] = v * int64(domain.state)
+		}
+		s.addUsage(domain.id, usage)
```

Comment:
I believe this will work fine for now as TAS is not combined with cohorts, and so only one workload is considered in each scheduling cycle on the snapshot. However, this will not work well when we start supporting cohorts - in that case usage coming from the assignment phase for a workload that is de-prioritized in a cohort will consume the capacity, but scheduler should operate on "clean" capacity.

One way to go about it in the future would be to remember usage coming from "Inflight" PodSets, and clean it at the end of the assignment phase, before the actual scheduling phase.

Since I don't see it problematic in the current assumptions (no cohorts) I believe a TODO with https://github.com/kubernetes-sigs/kueue/issues/3761 is sufficient.

Reply:
Will add tests later.

Reply:
Do you mean when having time, or in a follow up PR? My preference would be to add the test in this PR.

Reply:
I mean when I have time later, have codes without tests is terrible.

Reply:
Sure, I would like to include the fix in 0.10.1, and tentatively planning next week. Do you think you would add the test by then? I can also help to write it

Reply:
will do this weekend

Reply:
Can we have a test that covers a case where request topology is not the lowest level? E.g. the lowest level is hostname and job request (and can only fit in) a block? I have concerns if it's gonna work if the requested level is not the lowest one, since `s.addUsage()` method add usage only to leaves in the topology tree structure

Reply:
Make sense, but I think it still works because we'll traverse down the domains level by level until the lowest ones, which means we'll always get the lowest domains in `addUsage()`. Added the tests. PTAL.

Reply:
Can you adjust the test so the topology is made out of three levels. And then add also another podset that should make a workload not fit? I am afraid this usage may not be counted if it's not at the lowest level, so this is what I wanted to test

Reply:
After second consideration I think, you're actually right and the usage will be tracked correctly. Thanks for adding the test